src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -45,3 +60,80 @@ export const getDashboardForms = (
       })
   )
 }
+
+/**
+ * Private function to generate a presigned POST URL to upload into the given
+ * bucket.
+ *
+ * @param bucketName the name of the bucket to upload into
+ * @param param.fileId key of the file
+ * @param param.fileMd5Hash the MD5 hash of the file
+ * @param param.fileType the file type of the file
+ */
+const createPresignedPost = (
+  bucketName: string,
+  { fileId, fileMd5Hash, fileType }: PresignedPostParams,
+): ResultAsync<PresignedPost, InvalidFileTypeError | ExternalError> => {
+  if (!VALID_UPLOAD_FILE_TYPES.includes(fileType)) {
+    return errAsync(
+      new InvalidFileTypeError(`"${fileType}" is not a supported file type`),
+    )
+  }
+
+  const presignedPostPromise = new Promise<PresignedPost>((resolve, reject) => {
```

Comment:
have you tried `util.promisify`? it'll probably be cleaner than using a Promise constructor - something like
```
ResultAsync.fromPromise(
  promisify(AwsConfig.s3.createPresignedPost)({ /* arguments */ })
)
```

Reply:
i thought about it leh, but i also recall something wonky happening with promisify

Reply:
let me try

Reply:
yup, somehow the promise is not propagating to the callback. Doesn't work